### DISCUSSION TITLE
Polymer to Lit migration clarification for Enterprise Web App

### DISCUSSION THREAD

**Discussion by senthilcv** [lit/lit#2396] [Q&A] [answered]
Hi,

We have a Enterprise level Healthcare application that was in client server and 50% of it is converted to Web with Polymer 2.0. Before converting remaining application to web, we have a following dilemma.

Option 1. Do we just upgrade to polymer 3.0 and continue with rest of the conversion to Web and add Lit components on needed basis
or
Option 2: Convert current web application that is in polymer 2.0  to Lit and then convert rest of legacy client server to application to Web  using Lit

Our preference is option 1 but questions are raised about current polymer based web components not being supported in future browser updates.

Any insight or clarification will be really appreciated.

Current Web application has about 60 web components and 300 pages.

Thanks.
-Senthil

**Comment by kito99** [comment]
I think your best bet is (2). Polymer 3 is pretty much dead, unfortunately... If you choose (1) your "migrated" app will have way too much technical debt; it's also going to be harder to find support for Polymer 3 at this point. At least with Lit, it's much lighter weight and I dare say more broadly used... so your risk going forward is much lower should Google decide to abandon it.

**Reply by senthilcv** [reply]
Kito,

Thanks for your reply. Hope Google does not abandon it (Lit).

**Comment by justinfagnani** [comment, marked as answer]
I think the best approach is probably a hybrid of (1) and (2):

1. First, convert the existing project to Polymer 3. This is so that you'll be using JS modules and npm instead of HTML imports and Bower. Polymer 3 is 100% a mechanical translation of Polymer 2, so converting components should be relatively easy. The hardest part is probably mapping Bower dependencies to npm, but if you don't have many dependencies, it won't be that bad. You can try to use our "modulator" tool to do the conversion, though it's been a while since we maintained that.
2. Add new components with Lit and incrementally convert old component from Polymer 3 to Lit. Polymer and Lit components are interoperable, so you can do this at whatever pace you want. I would mostly focus on not writing new Polymer code, but the existing code will continue to work just fine.

**Reply by senthilcv** [reply]
Hi Justin,

Thanks for your reply and most likely that's route( hybrid) we will take.

-Senthil
